storage/src/vespa/storage/bucketdb/storagebucketdbinitializer.cpp

```diff
@@ -155,9 +165,14 @@ StorageBucketDBInitializer::onOpen()
         // Trigger bucket database initialization
     for (uint32_t i=0; i<_system._partitions.size(); ++i) {
         if (!_system._partitions[i].isUp()) continue;
-        ReadBucketList::SP msg(new ReadBucketList(BucketSpace::placeHolder(), spi::PartitionId(i)));
-        _state._lists[msg->getMsgId()] = msg;
-        sendDown(msg);
+        assert(_readState[i].get() != nullptr);
+        const BucketSpaceReadState &spaceState = *_readState[i];
+        for (const auto &stateElem : spaceState) {
+            document::BucketSpace bucketSpace = stateElem.first;
+            ReadBucketList::SP msg(new ReadBucketList(bucketSpace, spi::PartitionId(i)));
```

Comment:
Consider `auto msg = std::make_shared<ReadBucketList>(...` instead of naked `new`